src/lib/txn.js

```diff
@@ -195,23 +164,22 @@ const sendSignedTransaction = (web3, stx, doubtNonceError, confirmationCb) => {
 
 // returns a Promise<bool>, where the bool indicates tx success/failure
 const waitForTransactionConfirm = (web3, txHash) => {
-  return new Promise((resolve, reject) => {
-    const checkForConfirm = async () => {
-      console.log('checking for confirm', txHash);
-      const receipt = await web3.eth.getTransactionReceipt(txHash);
-      console.log('tried, got', receipt);
-      let confirmed = receipt !== null;
-      if (confirmed) resolve(receipt.status === true);
-      else setTimeout(checkForConfirm, CHECK_BLOCK_EVERY_MS);
-    };
-    checkForConfirm();
-  });
+  return retry(async (bail, n) => {
+    const receipt = await web3.eth.getTransactionReceipt(txHash);
+    let confirmed = receipt !== null;
```

Comment:
nit: `const confirmed`

Reply:
also, 'confirmed' isn't technically correct here, right? this is more of a 'was the transaction processed' or `hasReceipt` boolean, since whether or not the tx is confirmed is down to the `receipt.status` state.

Reply:
Pretty sure we still call that confirmed, even if status is false? A failed transaction still has an impact (gas cost), gets included in the chain, and thus gets confirmed with every block it's included in.